libs/device/feature/src/lib/restore/restore.service.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { Result, ResultObject } from "Core/core/builder"
+import { IpcEvent } from "Core/core/decorators"
+import { AppError } from "Core/core/errors"
+import { DeviceManager } from "Core/device-manager/services"
+import { DeviceId } from "Core/device/constants/device-id"
+import { ApiResponse } from "Core/device/types/mudita-os"
+import {
+  APIRestoreServiceEvents,
+  GeneralError,
+  PreRestore,
+  PreRestoreValidator,
+  Restore,
+  RestoreValidator200,
+  RestoreValidator202,
+} from "device/models"
+import { random } from "lodash"
+
+export class APIRestoreService {
+  constructor(private deviceManager: DeviceManager) {}
+
+  @IpcEvent(APIRestoreServiceEvents.PreRestore)
+  public async preRestore({
+    features,
+    deviceId,
+  }: {
+    features: string[]
+    deviceId?: DeviceId
+  }): Promise<ResultObject<PreRestore>> {
+    const device = deviceId
+      ? this.deviceManager.getAPIDeviceById(deviceId)
+      : this.deviceManager.apiDevice
+
+    if (!device) {
+      return Result.failed(new AppError(GeneralError.NoDevice, ""))
+    }
+
+    const restoreId = random(1, 100000)
```

Comment:
🟡 
Using `const restoreId = random(1, 100000);` poses a risk of ID collisions, especially in systems generating many IDs. For better uniqueness, consider using UUIDs, e.g., `import { v4 as uuidv4 } from 'uuid'; const restoreId = uuidv4();`. This approach significantly reduces collision risks.